src/map/skill.cpp

```diff
@@ -15825,15 +15845,30 @@ static int32 skill_dance_overlap_sub(struct block_list* bl, va_list ap)
 	return 1;
 }
 
-//Does the song/dance overlapping -> dissonance check. [Skotlex]
-//When flag is 0, this unit is about to be removed, cancel the dissonance effect
-//When 1, this unit has been positioned, so start the cancel effect.
-int32 skill_dance_overlap(struct skill_unit* unit, int32 flag)
+/**
+ * Does the song / dance overlapping->dissonance check.
+ * @param unit The unit that is about to be removed or was placed
+ * @param flag see enum e_dance_overlap
+ * @return Number of skill units that overlap with current skill unit and meet the overlap criteria (excluding current skill unit)
+ */
+int32 skill_dance_overlap(struct skill_unit* unit, e_dance_overlap flag)
```

Comment:
While you're at it, you could update the unit pointer to a reference

Reply:
Updated